core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,7 +64,23 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // Evaluate partitions in parallel. Partitions of each rdd will be cached by the `partitions`
+  // val in `RDD`.
+  private[spark] lazy val parallelPartitionEval: Boolean = {
```

Comment:
One `HadoopRDD` has one `InputFormat` to generate its splits/partitions, and one `UnionRDD` has many `HadoopRDD`s here. If each child `HadoopRDD`'s partitions are generated in parallel, individually they still compute their partitions serially as before. I think its access to its single instance of `InputFormat` doesn't become (more) concurrent. And in `getPartitions` it's only being used to compute splits rather than read data.

Either that's good news and there's no problem of this form or I have missed a point here?